src/posts/polyteos10-kernel/index.md

```diff
@@ -0,0 +1,29 @@
+---
+title: 'PolyTEOS10_bsq Kernel Bug Identified'
+date: '2025-08-05'
+authors:
+  - name: Michael Denes
+    github: michaeldenes
+summary: 'An issue with the `PolyTEOS10_bsq` kernel has been identified, where incorrect values of seawater density were being computed. In this post, we explain the issue and how it may impact your simulations.'
+---
+
+**Update to `PolyTEOS10_bsq` kernel**
+
+In recent days we’ve come across a bug in an application kernel of `parcels`. The `PolyTEOS10_bsq` kernel is used to calculate the density of seawater from the temperature and salinity fields. The kernel is based on equation (13) from
+[Roquet et al. (2014)](https://doi.org/10.1016/j.ocemod.2015.04.002) where the density $\rho$ is determined from the summation of a vertical reference profile $r_0$ and a residual function or density anomaly $r$. That is, $\rho(S,T,z) = r_0(z) + r(S,T,z)$.
+
+Up until now the kernel only computed the density anomaly, and was missing the vertical reference profile. At the ocean surface this isn’t a problem, as $r_0(z=0) = 0$ and so the computed density was correct. However, $r_0(z)>0$ below the ocean surface, and at relatively deep depths, this value can be significant. For example, at 100 m depth this value is 0.46443 kg/m$^3$ and at 500 m depth this value is 2.31175 kg/m$^3$, see the figure below.
+
+![Vertical reference profile as a function of depth](/posts/polyteos10-kernel/depth_vs_r0.png)
+
+By using the old kernel implementation, the seawater density at depths below the ocean surface were being underestimated, as the actual seawater density would have otherwise been denser. Simulations that computed vertical velocities (such as sinking/rise velocities) using density differences between a particle and its surrounding seawater may be impacted.
```

Comment:
```suggestion
By using the kernel implementation before the fix in PR [#2133](https://github.com/OceanParcels/Parcels/pull/2133), the seawater density at depths below the ocean surface were being underestimated, as the actual seawater density would have otherwise been higher. Simulations that computed vertical velocities (such as sinking/rise velocities) using density differences between a particle and its surrounding seawater may have been impacted.
```